lib/readline.js

```diff
@@ -45,3 +46,34 @@ module.exports = {
   emitKeypressEvents,
   moveCursor
 };
+
+let promises;
+
+Object.defineProperties(module.exports, {
+  promises: {
+    configurable: true,
+    enumerable: false,
+    get() {
+      if (promises === undefined) {
+        const {
+          Interface,
+          createInterface
+        } = require('internal/readline/promises');
+
+        promises = {
+          Interface,
+          clearLine,
+          clearScreenDown,
+          createInterface,
+          cursorTo,
+          emitKeypressEvents,
+          moveCursor
+        };
+
+        process.emitWarning('The readline.promises API is experimental',
+                            'ExperimentalWarning');
```

Comment:
I guess other people disagree, but I’m good with dropping the warning.

Reply:
I dropped the warning in a fixup commit. That way if a discussion ensues, it is isolated.